### PR TITLE
Feature: Object binding by annotation.

### PR DESCRIPTION
**Pull request by emhacker**

### CODE REVIEW

**Comment by emhacker** [comment]
Just a small quick improvement for the open source package.

File: query.py [hunk 1]
```diff
@@ -184,10 +184,10 @@ def usage_example():
     linus.add_child_node(QueryNode('id'))
     linus.add_child_node(QueryNode('email'))
     linus.add_child_node(QueryNode('avatarUrl')).add_arg('size', 20)
-    pwd = open('pwd.txt', 'r')
+    with open('pwd.txt', 'r') as f:
```
Comment (rantav) [inline query.py:187]: nitpick: I would pass the password simply as CLI arg.
not a big deal though

File: query_binder.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+class Decorator:
+    '''
+    Parameterized function decorator for scalar binding.
+    '''
+    def __init__(self, field, url, http):
+        self.field = field
+        self.url = url
+        self.http = http
+
+    def decorate(self, f):
```
Comment (rantav) [inline query_binder.py:14]: nice

File: query_binder.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+def bind(field, url, http):
+    d = Decorator(field, url, http)
+    return d.decorate
+
+
+def get_http():
+    session = requests.Session()
+    session.headers['Authorization'] = 'Bearer {}'.\
+        format('<your gh token here>')
```
Comment (rantav) [inline query_binder.py:35]: would make more sense to accept the token as an arg for `get_http`, right?

Comment (emhacker) [reply query_binder.py:35]: yes... Fixed.

**Comment by emhacker** [comment]
Still this code doesn't solves non-scalar queries elegantly.

**Comment by rantav** [comment]
What's the status with this one?

**Comment by rantav** [comment]
So, this is kind of old. Where does this stand?...

**Comment by emhacker** [comment]
Merged.

**Pull request merged**